consoles/console.pm

```diff
@@ -85,6 +86,29 @@ sub select {
     return $activated;
 }
 
+sub save_snapshot {
+    my ($self, $name) = @_;
+    $self->snapshot($name, 'activated', $self->{activated});
+}
+
+sub load_snapshot {
+    my ($self, $name) = @_;
+    $self->{activated} = $self->snapshot($name, 'activated') // 0;
+}
+
+sub snapshot {
```

Comment:
It's not really obvious what this function actually does based on the name and there's not a comment either.

The functionality is mostly split into `get` and `set` paths anyway,  so what about having separate functions for those?

Reply:
Sure I can split it. I had this `has` style in my mind like: https://metacpan.org/pod/Class::Accessor#SYNOPSIS
Actually I'm not sure what is the best practice in perl for `getter/setter` but I thought it is it. Do we still wanna have it splitted?

Also I'm not sure why we use `Class::Accessor` here and not `Mojo::Base`... but I think this is a topic for a other PR.

Reply:
The issue is that it's not just a plain getter/setter for an attribute, but is actually passed two parameters as well. That makes it look much more like a regular function call than getter or setter.

So IMO the code would be shorter and easier to understand by having separate methods.

And a step further, the "make a function if you do it more than once" rule can be applied in reverse: Both setter and getter are only called once, so it could even be inlined completey.

Reply:
ack, I will change that.